### ISSUE TITLE
Reduce output augmented diff filesize

### ISSUE THREAD

**Issue by CloudNiner** [azavea/onramp#30]
The output diffs are approximately 20GB worth of XML files for a single day when run on OSM. 

Extrapolating at current S3 costs for the monthly cost to store one year's worth of diffs:
20GB/day * 365 days/year * 0.023 $/GB-mo = 168 $/mo

I ran Gzip on a large 120mb diff which reduced the file size there to 22mb. Bzip2 reduced the same file to 18mb. 

The goal for this task is to reduce the output file size being stored on S3 by as much as we can without sacrificing easy compatibility with tools that would download and use these files. Gzip or Bzip2 are likely candidates but if there are others, they should be considered. Any solution needs to be able to operate within the one minute loop we have available before we need to begin writing the next diff.

We should be sure to correctly set metadata such as content-encoding for files directly uploaded to S3.

**Comment by hectcastro** [comment]
I think GZIP is going to be more universally supported (directly via browser or even some HTTP client interactions). This appears to be what AWS uses too for CloudTrail logs (which get dumped to S3). Also, depending on what you use to do the compression, it may have tunable settings for speed vs. compression:

> Regulate the speed of compression using the specified digit #, where -1 or --fast indicates the fastest compression method (less compression) and -9 or --best indicates the slowest compression method (best compression). The default compression level is -6 (that is, biased towards high compression at expense of speed).

(This is from the `gzip` man page.)

XML is expected to compress pretty well too, so those numbers make sense. Good call on setting `Content-Encoding` too. Adding that should help enable any automatic support in tooling referenced above.

**Comment by CloudNiner** [comment]
Good point. I had done some preliminary tests with the gzip tuning options but didn't add them before, here's the results on the same 120mb xml file as before, running on my MBP:
| gzip compression level | running time | output file size |
|---|---|---|
| 1 | 1.3s | 24mb |
| 6 (default) | 4.4s | 21mb |
| 9 | 19.3s | 20mb |

Given that we have to stay within the 1min loop, it seems reasonable to just continue using the default since these large files are outliers and we get most of the compression for a relatively low run time commitment at level 6. Turns out its a good default!

In summary -- we'll start with gzipping the xml before upload and just use the default gzip compression level.
